share/eds/store.go

```diff
@@ -419,3 +431,39 @@ func setupPath(basepath string) error {
 	}
 	return nil
 }
+
+// inMemoryOnceMount is used to allow reading once from buffer before using main mount.Reader
+type inMemoryOnceMount struct {
+	buf *bytes.Buffer
+
+	readOnce atomic.Bool
+	mount.Mount
+}
+
+func (m *inMemoryOnceMount) Fetch(ctx context.Context) (mount.Reader, error) {
+	if !m.readOnce.Swap(true) {
+		reader := &inMemoryReader{Reader: bytes.NewReader(m.buf.Bytes())}
```

Comment:
Why not pass Buffer instead of Reader?

Reply:
`bytes.Reader` implements `io.ReaderAt` and `io.Seeker` interfaces that required to satisfy `mount.Reader` interface. `bytes.Buffer` doesn't implement those. I didn't want to implement those interfaces, since it could be a bit complex while unnecessary.